pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -347,6 +347,9 @@ func (a *sqlInstanceAdapter) Delete(ctx context.Context, deleteOp *directbase.De
 	if err != nil {
 		return false, fmt.Errorf("deleting SQLInstance %s failed: %w", a.resourceID, err)
 	}
+	if err := a.pollForLROCompletion(ctx, op, "delete"); err != nil {
```

Comment:
I would have expected us to pass `deleteOp` into `pollForLROCompletion`

Reply:
We need a handle to the LRO to poll / wait. `deleteOp` does not provide that (currently). Probably some way to improve it in the future, but that would likely be a separate effort. We'd need to decide how to make the interface work for services that do not use LROs, and for each service's separate LRO implementation.

Reply:
Thanks. Please update this once you have the follow up LRO improvements